python/ray/data/block.py

```diff
@@ -60,35 +60,33 @@
 
 def _validate_key_fn(ds: "Dataset", key: KeyFn) -> None:
     """Check the key function is valid on the given dataset."""
-    try:
-        fmt = ds.dataset_format()
-    except ValueError:
+    schema = ds.schema(fetch_if_missing=True)
```

Comment:
I wonder if this can cause duplicated executions. Maybe it's fine since schema() only computes the first block.

Reply:
Yeah...the behavior should be the same as before with `dataset_format` also triggering execution.

Reply:
Ok. Sounds like something that will be solved with the new backend swap anyhow.